bpf/bpf_lxc.c

```diff
@@ -52,26 +53,28 @@ __BPF_MAP(CT_MAP4, BPF_MAP_TYPE_HASH, 0, sizeof(struct ipv4_ct_tuple),
 #if !defined DISABLE_PORT_MAP && defined LXC_PORT_MAPPINGS
 static inline int map_lxc_out(struct __sk_buff *skb, int l4_off, __u8 nexthdr)
 {
-	int csum_off = l4_checksum_offset(nexthdr);
+	struct csum_offset off = {};
 	uint16_t sport;
 	int i, ret;
 	struct portmap local_map[] = {
 		LXC_PORT_MAPPINGS
 	};
 
 	/* Ignore unknown L4 protocols */
-	if (unlikely(!csum_off))
+	if (nexthdr != IPPROTO_TCP && nexthdr != IPPROTO_UDP)
```

Comment:
This is in the port mapping code. We used the offset function to determine TCP/UDP or something else. As we now need checksum offsets for ICMPv6 I've made the check for TCP/UDP in port mapping explicit to allow using the same offset function for all of them. ICMP for east west is support.